osquery/tables/system/linux/smbios_tables.cpp

```diff
@@ -223,6 +223,25 @@ QueryData genMemoryDeviceMappedAddresses(QueryContext& context) {
   return results;
 }
 
+QueryData genOEMStrings(QueryContext& context) {
+  QueryData results;
+
+  LinuxSMBIOSParser parser;
+  if (!parser.discover()) {
+    return results;
+  }
+
+  parser.tables([&results](size_t index,
+                           const SMBStructHeader* hdr,
+                           uint8_t* address,
+                           uint8_t* textAddrs,
+                           size_t size) {
+    genSMBIOSOEMStrings(hdr, address, textAddrs, size, results);
```

Comment:
I see this being used on all other functions and I like consistency so this is fine. But I'm curious why are we creating a lambda here instead of using a `std::bind`?

Reply:
Hmm...You're right it does seem more semantically correct to use `std::bind` in this scenario.  I really didn't think about using it though.  Aside it from being used in similar scenarios in `smbios_tables.cpp`, I guess I've always preferred lambdas because they were easier to understand for me, also read they were more likely to be optimized by the compiler.  Although I'm hardly a very experienced `c++` user, so it easily could have been the incorrect assumption.